src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -306,63 +307,111 @@ def rsub(
         """
         return self.__apply_op(y, "sub")
 
-    def mul(self, y):
+    def mul(self, y: Union[int, float, torch.Tensor]) -> "ReplicatedSharedTensor":
         """Apply the "mul" operation between "self" and "y".
 
         Args:
             y: self*y
 
+        Returns:
+            ReplicatedSharedTensor: Result of the operation.
+
+        Raises:
+            ValueError: Raised when private mul is performed parties!=3.
+
         """
+        y_tensor, session_vals = self.sanity_checks(self, y)
+        session, ring_size, config, rank, nr_parties = session_vals
+        is_private = isinstance(y, ReplicatedSharedTensor)
+
+        if is_private:
+            # currently there is no way of getting session of RStensor since session is optional.
```

Comment:
TYPO: ```RSTTEnsor```

Reply:
There should be a way to get it, right? (because you serialize the ```session_uuid``` and you get that?)

Reply:
Oh, I have to remove the comment. I did get the session and check for a number of parties.